controllers/posts_controller.js

```diff
@@ -5,16 +5,35 @@ const qs = require('querystring')
 
 module.exports.addPosts = (req, res) => {
     var userID = req.user.id
-    post.addPosts(req.body.description, req.body.url, req.body.tags, req.user.id, function (err, data) {
-        if(err)
-            return res.status(400).json({error: true, data: null})
-        user.linkPosts(userID, data, function (err, data) {
-            res.status(err ? 404 : 200).json({
-                error: err,
-                data: data
+    var url = req.body.url
+
+    var errorMessage = ""
+
+    if (utils.isEmpty(userID)) {
```

Comment:
I remember us talking about this `isEmpty`, but I feel it was more of a "educational" thing rather than something to do in practice. I feel like just doing `userId === ""` would be more straight forward.

Triple-equal is important because we don't want to assert type (which is done with double-equal)